tools/StaticAnalysis/ExampleAnalyzer/utils.ps1

```diff
@@ -114,13 +108,18 @@ function Get-ExamplesDetailsFromMd {
             # If there is no ```output``` split codelines and outputlines
             if ($exampleOutputBlocks.Count -eq 0) {
                 foreach ($exampleCodeBlock in $exampleCodeBlocks) {
-                    $codeRegex = "\n(([A-Za-z \t])*(PS|[A-Za-z]:)(\w|[\\/\[\].\- ])*(>|&gt;)+( PS)*)*[ \t]*((([A-Za-z]\w+-[A-Za-z]\w+\b(.ps1)?(?!(-|   +\w)))|(" +
-                    "(@?\((?>\((?<pair>)|[^\(\)]+|\)(?<-pair>))*(?(pair)(?!))\) *[|.-] *\w)|" +
-                    "(\[(?>\[(?<pair>)|[^\[\]]+|\](?<-pair>))*(?(pair)(?!))\]\$)|" +
-                    "(@{(?>{(?<pair>)|[^{}]+|}(?<-pair>))*(?(pair)(?!))})|" +
-                    "('(?>'(?<pair>)|[^']+|'(?<-pair>))*(?(pair)(?!))' *[|.-] *\w)|" +
-                    "((?<!``)`"(?>(?<!``)`"(?<pair>)|[\s\S]|(?<!``)`"(?<-pair>))*(?(pair)(?!))(?<!``)`" *[|.-] *\w)|" +
-                    "\$))(?!\.)([\w-~``'`"$= \t:;<>@()\[\]{},.+*/|\\&!?%#]*[``|][ \t]*(\n|\r\n)?)*([\w-~``'`"$= \t:;<>@()\[\]{},.+*/|\\&!?%#]*(?=\n|\r\n|#)))"
+                    $codeRegex = "\r?\n("+
+                    "(.*ForEach-Object {((.*\r?\n)|(\w*=.*))*\s*})|"+
+                    "((.*[A-Za-z]\w+-[A-Za-z]\w+\s*(``\r?\n)?)((\s*-.*``\r?\n)*(.*@{((.*\r?\n)|(\w*=.*))*\s*}\s*``?)+)+(?=\n|\r\n))|"+
+                    "((([A-Za-z \t])*(PS|[A-Za-z]:)(\w|[\\/\[\].\- ])*(>|&gt;)+( PS)*)*[ \t]*((([A-Za-z]\w+-[A-Za-z]\w+\b(.ps1)?(?!(-|   +\w)))|(" +
+                    "(@?\((?>\((?<pair>)|[^\(\)]+|\)(?<-pair>))*(?(pair)(?!))\) *[|.-] *\w)|" + # match ()
+                    "(\[(?>\[(?<pair>)|[^\[\]]+|\](?<-pair>))*(?(pair)(?!))\]\$)|" + # match []
+                    "((\s*-\w*\s*(``\r?\n\s*)?)?(\$\w*\s*=.*)?@{(?>{(?<pair>)|[^{}]+|}(?<-pair>))*(?(pair)(?!))})|" + # match @{}
+                    "('(?>'(?<pair>)|[^']+|'(?<-pair>))*(?(pair)(?!))' *[|.-] *\w)|" + # match ''
+                    "((?<!``)`"(?>(?<!``)`"(?<pair>)|[\s\S]|(?<!``)`"(?<-pair>))*(?(pair)(?!))(?<!``)`" *[|.-] *\w)|" + # match ""
+                    "(\$\w*\s*=.*)?(@`"\s*\r?\n?(\{\s*)?)(.*\r?\n)*((\s*\})?\s*\r?\n?`"@)|"+ # match @" "@
+                    "\$))(?!\.)([\w-~``'`"$= \t:;<>@()\[\]{},.+*/|\\&!?%#]*[``|][ \t]*(\n|\r\n)?)*([\w-~``'`"$= \t:;<>@()\[\]{},.+*/|\\&!?%#]*(?=\n|\r\n|#))))"+
```

Comment:
Can all `\n|\r\n` be replaced by `\r?\n`?

Reply:
Yes, I have replaced ```\r?\n``` to ```(\n|\r\n)```